### PR TITLE
[SPARK-47814][DSTREAM] Move `KinesisTestUtils` & `WriteInputFormatTestDataGenerator` from `main` to `test`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to move 
- the file `KinesisTestUtils.scala` 
  from: 'connector/kinesis-asl/src/`main`/scala/org/apache/spark/streaming/kinesis' 
  to: 'connector/kinesis-asl/src/`test`/scala/org/apache/spark/streaming/kinesis'
- the file `WriteInputFormatTestDataGenerator.scala`
 from: 'core/src/`main`/scala/org/apache/spark/api/python'
 to: 'core/src/`test`/scala/org/apache/spark/api/python'

### Why are the changes needed?
The logic related to `testing` should be placed in the dir `test` instead of the dir `main`.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @dongjoon-hyun @LuciferYang

**Comment by LuciferYang** [comment]
I suggest creating a Jira for this

**Comment by panbingkun** [comment]
> I suggest creating a Jira for this

Okay.

**Comment by panbingkun** [comment]
Wait, I found other classes with similar issues. Let me handle them all at once.
image

**Comment by panbingkun** [comment]
> > > Wait, I found other classes with similar issues. Let me handle them all at once. image
> > 
> > 
> > There are some classes that are just called `**Test`, they may not necessarily be suitable for placement in the `test` directory, such as `ANOVATest`, `ChiSquareTest` and `FValueTest`, in fact, they are used by `UnivariateFeatureSelector`.
> > also cc @zhengruifeng
> 
> yes, those ml algorithms' names end with `Test` but they are not real testsuites.

@LuciferYang @zhengruifeng 
Yes, I am checking and confirming one by one.

File: project/MimaExcludes.scala [hunk 1]
```diff
@@ -88,7 +88,12 @@ object MimaExcludes {
     // SPARK-47706: Bump json4s from 3.7.0-M11 to 4.0.7
     ProblemFilters.exclude[IncompatibleResultTypeProblem]("org.apache.spark.sql.expressions.MutableAggregationBuffer.jsonValue"),
     ProblemFilters.exclude[IncompatibleMethTypeProblem]("org.apache.spark.sql.types.DataType#JSortedObject.unapplySeq"),
-    ProblemFilters.exclude[IncompatibleMethTypeProblem]("org.apache.spark.mllib.tree.model.TreeEnsembleModel#SaveLoadV1_0.readMetadata")
+    ProblemFilters.exclude[IncompatibleMethTypeProblem]("org.apache.spark.mllib.tree.model.TreeEnsembleModel#SaveLoadV1_0.readMetadata"),
+    // SPARK-47814: Move `WriteInputFormatTestDataGenerator.scala` from `main` to `test`
```
Comment (dongjoon-hyun) [inline project/MimaExcludes.scala:92]: This seems to outdated. Could you match this with the PR title, @panbingkun ?

**Review by dongjoon-hyun — COMMENTED** [review]
It's a nice catch.

- I removed `[TESTS]` from the title because the original file is under `main`.
- Given the above and MIMA, we need a proper deprecation process in `branch-3.5` although this is a test helper.

In any way, it's great to find and fix before Apache Spark 4.0.0 release. I guess 4.0 is the only version we can remove this without any concern.

File: project/MimaExcludes.scala (hunk 1 above)
Comment (panbingkun) [reply project/MimaExcludes.scala]: Done

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by panbingkun** [comment]
> Merged to master for Apache Spark 4.0.0.
> 
> Could you make a deprecation PR to branch-3.5, @panbingkun ?

Sure, let me to do it.

**Comment by LuciferYang** [comment]
`WriteInputFormatTestDataGenerator` is called by `test_readwrite.py`. Even though the CI has passed, I still want to confirm that putting it into test.jar will not have a negative impact on the tests of pyspark.

https://github.com/apache/spark/blob/de2079163270d371e0ead33f09d61d66ee496e88/python/pyspark/tests/test_readwrite.py#L26-L31

**Comment by panbingkun** [comment]
> `WriteInputFormatTestDataGenerator` is called by `test_readwrite.py`. Even though the CI has passed, I still want to confirm that putting it into test.jar will not have a negative impact on the tests of pyspark. @panbingkun @dongjoon-hyun
> 
> https://github.com/apache/spark/blob/de2079163270d371e0ead33f09d61d66ee496e88/python/pyspark/tests/test_readwrite.py#L26-L31

Let me confirm, just a moment.

**Comment by panbingkun** [comment]
Currently, I am verifying locally that it is OK, as follows:
image

image

**Comment by LuciferYang** [comment]
> @LuciferYang Currently, I am verifying it on local, it is OK, as follows: image
> 
> image

Got ~ Thanks @panbingkun

**Pull request closed without merging**